### ISSUE TITLE
How could I view results table for multiple models?

### ISSUE THREAD

**Issue by davideuler** [carlini/yet-another-applied-llm-benchmark#14]
The project is every practical and helpful compared to lots of simple benchmarks.

I've tested several models, including the following models. 
Nous-Hermes-2-Mixtral-8x7B-DPO and bagel-dpo-34b-v0.2 has the highest pass rate among these models.

After I run a second test, the evaluation_examples result page shows only the latest model's result.
How could I view all models result which has been evaluated?

![image](https://github.com/carlini/yet-another-applied-llm-benchmark/assets/377983/184ab7d8-79c5-4626-a572-81729d0f3ece)

ps, here is the model which I tested. And I'm testing other model.
python main.py --model bagel --run-tests --generate-report  # 17%, bagel-dpo-34b-v0.2
python main.py --model laser-dolphine --run-tests --generate-report  # 10%
python main.py --model noushermes2 --run-tests --generate-report  # 15%
python main.py --model noushermes2-mix --run-tests --generate-report  # 20%, q5, Nous-Hermes-2-Mixtral-8x7B-DPO-GGUF
python main.py --model Nous-Capybara-limarpv3 --run-tests --generate-report  # 10%

**Comment by carlini** [comment]
Oh I should probably document this! (Or if you'd like to submit a PR with the explanation happy to take it.) All you have to do is run

```
python main.py --load-saved --model model1 --model model2 --model model3 --generate-report
```

and it will load the saved results and generate a table for you.

If you run with `--times 5` then you can get 5 runs for each configuration, or you can run with `--runid 0`, `--runid 0` etc. if you want to explicitly control each time you invoke the model.

**Comment by davideuler** [comment]
> Oh I should probably document this! (Or if you'd like to submit a PR with the explanation happy to take it.) All you have to do is run
> 
> ```
> python main.py --load-saved --model model1 --model model2 --model model3 --generate-report
> ```
> 
> and it will load the saved results and generate a table for you.
> 
> If you run your test case runs with `--times 5` then you can get 5 runs for each configuration, or you can run with `--runid 0`, `--runid 1` etc. if you want to explicitly control each time you invoke the model.

Thanks. That's cool. I'll submit a PR plus some fix and updates to the repo. 

The evaluation pipeline implementation is elegant and beautiful. I love it.
Currently, the benchmark in this repo is the only one which I trust for coding evaluation.  The tests are daily programming problem in real work situations. I tested models manually before I came across this benchmark.

If we can share a leaderboard, maybe more people would find it useful.

**Comment by davideuler** [comment]
![image](https://github.com/carlini/yet-another-applied-llm-benchmark/assets/377983/83cb8af9-d31c-46fa-abfa-e167b6926ac0)

So far, the 2 mixtral-8x7b model has the best performance. And then bagel-dpo-34b-v0.2.

**Comment by carlini** [comment]
Fascinating. Nice to have some open models here!

One thing I don't really know is what the margins of error are here. (Specifically: two things. It might be nice to show a +/- just for the statistical confidence given N runs per model per test. But more importantly and much harder to evaluate, I don't have a good idea if one model being +3% better "means" anything. That's like two tests. They could have been worded slightly differently, or some other test could have been added. I think swings of +/- 10% are probably meaningful, but I don't know where the point is to be able to say "a > b".)

**Comment by davideuler** [comment]
Agree. I've no clue to verify a model performs better with a difference of +3%.  +/- 10% seems to be more meaningful.  The point may be evaluated on more tests and comparison.

If run the test N times per model per test, it should be more fair. And maybe with some different parameters.
I tested the case of program_pipes_python.py on GPT-4, even in GPT-4, most of time it failed on the case. I tested about 5 times, and only one run I get the correct anwser.
